packages/protocol/src/interaction/DecodedDataReport.ts

```diff
@@ -5,25 +5,44 @@
  */
 
 import { DataReport } from "#types";
-import { DecodedAttributeReportValue, normalizeAndDecodeReadAttributeReport } from "./AttributeDataDecoder.js";
-import { DecodedEventReportValue, normalizeAndDecodeReadEventReport } from "./EventDataDecoder.js";
+import {
+    DecodedAttributeReportStatus,
+    DecodedAttributeReportValue,
+    normalizeAndDecodeReadAttributeReport,
+} from "./AttributeDataDecoder.js";
+import {
+    DecodedEventReportStatus,
+    DecodedEventReportValue,
+    normalizeAndDecodeReadEventReport,
+} from "./EventDataDecoder.js";
 
 export interface DecodedDataReport extends DataReport {
     isNormalized: true;
     attributeReports: DecodedAttributeReportValue<any>[];
+    attributeStatus?: DecodedAttributeReportStatus[];
     eventReports: DecodedEventReportValue<any>[];
+    eventStatus?: DecodedEventReportStatus[];
 }
 
 export function DecodedDataReport(report: DataReport): DecodedDataReport {
-    if ((report as DecodedDataReport).isNormalized) {
-        return report as DecodedDataReport;
+    if ((report as unknown as DecodedDataReport).isNormalized) {
```

Comment:
Eww, why was this necessary?  TS bug?

Reply:
think was leftover from my trying something else in devrelopment. removed